crates/ironrdp-pdu/src/geometry.rs

```diff
@@ -206,10 +254,10 @@ impl<'de, I: RectangleImpl> PduDecode<'de> for BaseRectangle<I> {
 // TODO(@pacmancoder) this should be removed later
 mod legacy {
     use super::*;
+    use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+    use std::io;
 
-    pub type Rectangle = InclusiveRectangle;
-
-    impl Rectangle {
+    impl InclusiveRectangle {
         // TODO: clarify code related to rectangles (inclusive vs exclusive bounds, …)
         // See for instance:
         // https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/84a3d4d2-5523-4e49-9a48-33952c559485
```

Comment:
I guess this TODO will be addressed by your changes?

Reply:
Yes, PR is still in draft sadly, my refactoring broke a bit more of a code than expected

Reply:
No problem!

Reply:
Well, this happens to be a rabbit hole... To properly refactor this, a huge amount of code (`rectangle_processing.rs`) should be rewritten to use generic `Rectangle` instead of concrete `InclusiveRectangle` and validated/tested properly, because currently `rectangle_processing.rs` is a huge 1600 LoC file with mixed code for both exclusive and inclusive rectangles...

I propose to leave this as is (ExplicitRectangle used as InclusiveRectangle for operations) but create an issue because It looks like is way out of scope currently :(

Reply:
Sure, let’s keep all that as a future improvement! Feel free to create an issue on GitHub with the "improvement" label 👍